pi.py

```diff
@@ -1 +1,578 @@
+            #
+            
+            # set loop counter and initial condition
+            NC=0
+            TG=0
+
+            # loop until loop converges or bails out
+            while ((np.abs(TGNEW-TG)) > 0.001):
+            
+                # Parcel temperature and mixing ratio during this iteration
+                TG=TGNEW
+                TC=TG-273.15
+                ENEW=6.112*np.exp(17.67*TC/(243.5+TC))
+                RG=EPS*ENEW/(P[j]-ENEW)
+                
+                # increase iteration count in the loop
+                NC=NC+1
+                
+                #
+                #   ***  Calculate estimates of the rates of change of the entropy    ***
+                #   ***           with temperature at constant pressure               ***
+                #
+
+                ALV=ALV0+CPVMCL*TC
+                SL=(CPD+RP*CL+ALV*ALV*RG/(RV*TG*TG))/TG
+                EM=RG*P[j]/(EPS+RG)
+                # (last term vanishes with saturation, i.e. RH=1)
+                SG=(CPD+RP*CL)*np.log(TG)-RD*np.log(P[j]-EM)+ALV*RG/TG
+                # convergence speed (AP, step in entropy fraction) varies as a function of 
+                # number of iterations
+                if (NC < 3):
+                    # converge slowly with a smaller step
+                    AP=0.3
+                else:
+                    # speed the process with a larger step when nearing convergence
+                    AP=1.0
+                # find the new temperature in the iteration
+                TGNEW=TG+AP*(S-SG)/SL
+                
+                #
+                #   ***   If the routine does not converge, set IFLAG=2 and bail out   ***
+                #
+                if (NC > 500) or (ENEW > (P[j]-1)):
+                    CAPED=0
+                    TOB=T[0]
+                    LNB=P[0]
+                    IFLAG=2
+                    # Return the uncoverged values
+                    return(CAPED,TOB,LNB,IFLAG)
+                
+                # store the number of iterations
+                NCMAX=NC
+                
+            #
+            #   *** Calculate buoyancy   ***
+            #
+            # Parcel Mixing ratio: either reversible (ascent_flag=0) or pseudo-adiabatic (ascent_flag=1)
+            RMEAN=ascent_flag*RG+(1-ascent_flag)*RP
+            # Parcel and Environmental Density Temperatures at this pressure (E94, EQN. 4.3.1 and 6.3.7)
+            TLVR=TG*(1.+RG/EPS)/(1.+RMEAN)
+            TENV=T[j]*(1.+R[j]/EPS)/(1.+R[j])
+            # Bouyancy of the parcel in the environment (Proxy of E94, EQN. 6.1.5)
+            TVRDIF[j,]=TLVR-TENV
+            
+
+    #
+    #  ***  Begin loop to find Positive areas (PA) and Negative areas (NA) ***
+    #                  ***  and CAPE from reversible ascent ***
+    NA=0.0
+    PA=0.0
+    
+    #
+    #   ***  Find maximum level of positive buoyancy, INB    ***
+    #
+    INB=0
```

Comment:
Just my convention style, but whenever something is an index, I always tack on a "_idx" at the end.

Reply:
I may update this to use the "j" convention the rest of the script relies on.